h2/src/main/org/h2/mvstore/MVStore.java

```diff
@@ -1781,67 +1786,67 @@ private void compactMoveChunks(Iterable<Chunk> move) {
             // now re-use the empty space
             reuseSpace = true;
             for (Chunk c : move) {
-                moveChunk(c, false);
+                moveChunk(c, 0);
             }
 
             // update the metadata (within the file)
             commit();
             sync();
-            if (moveChunk(chunk, false)) {
+            if (moveChunk(chunk, 0)) {
                 commit();
```

Comment:
Here, also the committing chunk may overwrite the last moved chunk.

Reply:
Do we care here? Even if it overwrites previous chunks positions - we already saved a new meta map with latest chunks positions, so it should be ok.

Reply:
No, no, you've misunderstood. Line 1796, the committing chunk overwrites the free space of the moved chunk in the line 1795, and at the moment, the committing is persisting,  but not sync().

Reply:
Yep, you are right! If it crashes in the middle of that commit write, but already overwritten previous chunk location - we have a corruption.

Reply:
Same here, with last commit overlap should not occur

Reply:
@andreitokar I think that your algorithm is complicated, and some bug exist yet.

Reply:
I am working on them... :smile: